# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoadRequest.h
# Commit: f0d873f64979
# Full Hash: f0d873f64979928a059b519ea8409dd3c5286d39
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-10-03 20:53 +0000
# Regressor Bug: 1968890
# File Overlap Count: 5
# Overlapping Files: js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp, js/loader/ModuleLoadRequest.h, js/loader/ModuleLoaderBase.h, dom/script/ScriptLoader.cpp
# Description:
#   Bug 1968890 - Part 7: Add Cancel method. r=jonco
# ==============================================================================

diff -r f06186536887 -r f0d873f64979 js/loader/ModuleLoadRequest.h
--- a/js/loader/ModuleLoadRequest.h	Fri Oct 03 20:53:29 2025 +0000
+++ b/js/loader/ModuleLoadRequest.h	Fri Oct 03 20:53:29 2025 +0000
@@ -66,7 +66,7 @@
   nsIGlobalObject* GetGlobalObject();
 
   void SetReady() override;
-  void Cancel() override;
+  void Cancel() override { mLoader->Cancel(this); };
 
   void SetImport(Handle<JSScript*> aReferrerScript,
                  Handle<JSObject*> aModuleRequestObj, Handle<Value> aPayload);